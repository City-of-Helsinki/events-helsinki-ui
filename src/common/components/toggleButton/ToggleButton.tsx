import classNames from "classnames";
import React, { MutableRefObject } from "react";

import styles from "./toggleButton.module.scss";

interface Props {
  buttonRef?: MutableRefObject<HTMLButtonElement | null>;
  icon?: React.ReactElement;
  isSelected: boolean;
  onClick: (value: string) => void;
  testId?: string;
  text: string;
  value: string;
}

const ToggleButton: React.FC<Props> = ({
  buttonRef,
  icon,
  isSelected,
  onClick,
  testId,
  text,
  value
}) => {
  const handleClick = () => {
    onClick(value);
  };
  return (
    <button
      ref={buttonRef}
      data-testid={testId}
      className={classNames(styles.toggleButton, {
        [styles.isSelected]: isSelected
      })}
      aria-pressed={isSelected}
      onClick={handleClick}
      type="button"
    >
      <div className={styles.innerWrapper}>
        {icon}
        <span>{text}</span>
      </div>
    </button>
  );
};

export default ToggleButton;
