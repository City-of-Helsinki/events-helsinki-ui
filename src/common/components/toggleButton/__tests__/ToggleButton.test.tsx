import * as React from "react";
import renderer from "react-test-renderer";

import ToggleButton from "../ToggleButton";

test("ToggleButton matches snapshot", () => {
  const component = renderer.create(
    <ToggleButton
      isSelected={true}
      onClick={() => {}}
      text={"Test button"}
      value={"test"}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

export {};
