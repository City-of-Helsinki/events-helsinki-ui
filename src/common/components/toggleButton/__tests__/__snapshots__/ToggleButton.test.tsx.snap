// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton matches snapshot 1`] = `
<button
  aria-pressed={true}
  className="toggleButton isSelected"
  onClick={[Function]}
  type="button"
>
  Test button
</button>
`;
